std/array.d

```diff
@@ -3008,11 +3008,17 @@ if (isDynamicArray!A)
         this.arr = arr;
     }
 
-    auto opDispatch(string fn, Args...)(Args args) if (is(typeof(mixin("impl." ~ fn ~ "(args)"))))
+    template opDispatch(string fn)
     {
-        // we do it this way because we can't cache a void return
-        scope(exit) *this.arr = impl.data;
-        mixin("return impl." ~ fn ~ "(args);");
+        // This is a different name plus an alias to work around issue 14604.
+        auto opDispatchImpl(Args...)(Args args)
```

Comment:
The original code doesn't work because `std.traits.hasMember` returns false when given an opDispatch like that. I checked with dmd 2.054 (the oldest I have at hand) and it does the same, so I don't think this is a regression.

However, I just found out that `__traits(hasMember, ...)` has no problem with such an opDispatch. So maybe `std.traits.hasMember` should be changed instead.